pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -108,13 +120,13 @@ func (c NodeController) sync(ctx context.Context, syncCtx factory.SyncContext) e
 	}
 
 	oldStatus := &operatorv1.StaticPodOperatorStatus{}
-	_, updated, updateError := v1helpers.UpdateStaticPodStatus(c.operatorClient, v1helpers.UpdateStaticPodConditionFn(newCondition), func(status *operatorv1.StaticPodOperatorStatus) error {
-		status.NodeStatuses = newTargetNodeStates
-		return nil
-	}, func(status *operatorv1.StaticPodOperatorStatus) error {
```

Comment:
this was at the end of the chain, so it received the status object that has been mutated already by the previous `UpdateStaticPodStatusFunc` funcs. `oldStatus` is basically the new status we are trying to save and it will have `newCondition`, right? 

So we may not see the event "MasterNodesReadyChanged" I think.

Reply:
I concur.

Reply:
yes, it seems it should have been at the beginning of the chain.

Reply:
or could we simply relay on `originalOperatorStatus` to detect changes?

Reply:
@p0lyn0mial are you suggesting we change the signature of the `UpdateStatus` function? I am in favor of changing the signature of the function (we have to add a v2 of this package I assume to maintain backward compatibility?)

Reply:
I was thinking about reading the old status from [originalOperatorStatus](https://github.com/openshift/library-go/blob/fe935f679430bcf7dfee61ddf7bec7c37db79db4/pkg/operator/staticpod/controller/node/node_controller.go#L42) but it seems to be not enough since we need to update both `newCondition` and `NodeStatus`

No need to create a new function (`v2`) as we don't have any guarantees of backward compatibility for `library-go`